group_utils/javascript/automation/lib/javascript/community/groupUtils.js

```diff
@@ -0,0 +1,347 @@
+/**
+ * Utilities for working with openHAB groups in JavaScript.
+ * This code is not compatible with the GraalVM JavaScript add-on.
+ *
+ * Copyright (c) 2021 Florian Hotze under MIT License
+ * @florian-h05(https://github.com/florian-h05)
+ */
+
+(function (context) {
+  /**
+   * Imports.
+   */
+  var logger = Java.type('org.slf4j.LoggerFactory').getLogger('org.openhab.model.script.Rules.GroupUtils')
+  var Collectors = Java.type('java.util.stream.Collectors')
+
+  /**
+   * Private function to get the direct members of a group.
+   *
+   * @param {string} group Name of the openHAB group
+   * @param {string} characteristic Which characteristic of the members you get, valid: name (default), label, state
+   * @returns {Array} Given characteristic of direct members
+   */
+  function getMembers (group, characteristic) {
+    logger.debug('Getting direct members of group ' + group)
```

Comment:
Each full line of code should end with a `;`. I've lost the long article I read about why but there are reasons.

Reply:
Do you use any code-linter?
If you used any, could you please share your settings?

Reply:
No, but I probably should. I think there is one built into VSCode, or can be installed as an extension but I just did it by eye here.

Reply:
Okay,

I used [standard](https://github.com/standard/standard) but because of the semicolons I will switch to [semistandard](https://github.com/standard/semistandard).